src/Symfony/Component/FeatureToggle/FeatureChecker.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureToggle;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    public function __construct(
+        private readonly FeatureCollection $features,
+        private readonly bool $whenNotFound,
+    ) {
+    }
+
+    public function isEnabled(string $featureName): bool
```

Comment:
can't we imagine adding an optional context here?
this logic of computation reminds me in some way the `workflow apply` logic

Reply:
We gave it a thought based on Serializer. ATM there is no need for a context. There will be when we introduce the "allow only for 10% of users" because we will need to calculate some IDs per request. Open to discuss )(either now if it must be part of the initial release or later if not)